### PR TITLE
rose bush: add rotated log files to the "cylc files" menu

### PR DESCRIPTION
**Pull request by oliver-sanders**
This adds a submenu to the "cylc files" menu listing all log file (newest at the top). If there are no older log files it displays as before.

To be merged before https://github.com/cylc/cylc/pull/1958.

@benfitzpatrick Please review
@arjclark Please review

### CODE REVIEW

File: lib/html/static/css/rose-bush.css [hunk 1]
```diff
@@ -41,5 +41,67 @@ pre {
      }
     .page_entries select {
         width: 300px;
-     } 
+     }
+}
+
+
+/* dropdown-submenu code from bootstrap v2.3.1 (removed in v3.x)
```
Comment (benfitzpatrick) [inline lib/html/static/css/rose-bush.css]: Probably best to add a note in ACKNOWLEGEMENT.md as well.

File: lib/python/rose/suite_engine_procs/cylc.py [hunk 2]
```diff
@@ -752,15 +752,44 @@ def get_suite_logs_info(self, user_name, suite_name):
...
+                    link_path = f_name
+                old_logs = []  # Old log naming system.
+                new_logs = []  # New log naming system.
+                # TODO: Post migration to cylc this logic can be replaced by:
+                # `from cylc.suite_logging import get_logs` (superior)
+                for log in glob(f_name + '.*'):
+                    log_name = os.path.basename(log)
+                    if log_name == link_path:
+                        continue
+                    if (len(log_name.split('.')) > 1 and
```
Comment (benfitzpatrick) [inline lib/python/rose/suite_engine_procs/cylc.py]: First condition should always be true due to the `glob`

**Comment by benfitzpatrick** [comment]
2 comments, otherwise looks good. Can we also have an enhancement to serve the `log.1`, etc, files in their own web page rather than as a download?

**Comment by oliver-sanders** [comment]
@benfitzpatrick Comments addressed.

File: lib/html/static/css/rose-bush.css (hunk 1 above)
Comment (oliver-sanders) [inline lib/html/static/css/rose-bush.css]: Done.

**Comment by benfitzpatrick** [comment]
> Can we also have an enhancement to serve the log.1, etc, files in their own web page rather than as a download?

(Talked with @oliver-sanders - this is actually fine for e.g. `log.20150101T0000Z` type files, but not for the old/current `log.1` file naming, due to `mimetypes.guess_type` functionality. No enhancement needed.)

File: ACKNOWLEDGEMENT.md [hunk 3]
```diff
@@ -66,6 +66,11 @@ lib/html/static/js/bootstrap.min.js:
   released under the Apache 2.0 license.
   See <http://getbootstrap.com/>.
 
+lib/html/static/css/rose-bush.css
+* Contains modified code from Twitter Bootstrap v2.3.1 released
+  under the Arache 2.0 licence.
```
Comment (benfitzpatrick) [inline ACKNOWLEDGEMENT.md]: Apache

Comment (oliver-sanders) [inline ACKNOWLEDGEMENT.md]: Done.

**Comment by benfitzpatrick** [comment]
OK by me, over to @arjclark

**Comment by arjclark** [comment]
Clicking on an older log (e.g. log.1) results in a download prompt being fired off. Can we not just natively view the file in the browser?

**Comment by benfitzpatrick** [comment]
See my comment higher up!

**Comment by arjclark** [comment]
@benfitzpatrick - tested and seen as working then for the log.<DATEPOINT> type files?

**Comment by benfitzpatrick** [comment]
Yep

**Comment by arjclark** [comment]
Looks ok to me. Test battery passing in my environment against cylc 6.10.2

**Pull request merged**